src/snowflake/snowpark/mock/connection.py

```diff
@@ -370,6 +377,24 @@ def execute(
 
         res = execute_mock_plan(plan)
         if isinstance(res, TableEmulator):
+            # stringfy the variant type in the result df
+            for col in res.columns:
+                # TODO: the first check is due to window function doesn't set res.sf_types[col]
```

Comment:
What's the first check? Is this outdated?

Reply:
oh yes this commenet it out of date, let me remove it